src/history/hs_rec.c

```diff
@@ -680,16 +646,25 @@ __wt_hs_insert_updates(
 }
 
 /*
- * __wt_hs_delete_key_from_ts --
- *     Delete history store content of a given key from a timestamp.
+ * __wt_hs_delete_reinsert_key_from_ts --
+ *     Delete history store content of a given key from a timestamp and optionally reinsert them
+ *     with ts-1 timestamp.
  */
 int
-__wt_hs_delete_key_from_ts(WT_SESSION_IMPL *session, WT_CURSOR *hs_cursor, uint32_t btree_id,
-  const WT_ITEM *key, wt_timestamp_t ts, bool reinsert)
+__wt_hs_delete_reinsert_key_from_ts(WT_SESSION_IMPL *session, WT_CURSOR *hs_cursor,
```

Comment:
I think there is no need for word `reinsert` in the name. I believe the function arguments are self explanatory.